public/API/API_GetGameList.php

```diff
@@ -91,15 +116,19 @@
 }
 
 if ($withHashes) {
+    $responseIndex = [];
+    foreach ($response as $index => $entry) {
+        $responseIndex[$entry['ID']] = $index;
+    }
+
     $hashes = GameHash::compatible()
         ->select('game_id', 'md5')
-        ->whereIn('game_id', array_column($response, 'ID'));
+        ->whereIn('game_id', array_column($response, 'ID'))
```

Comment:
Could use `array_keys($responseIndex)` here. Not sure if that would provide any additional performance avoiding the extra lookup.